src/engine/threaded_engine_perdevice.cc

```diff
@@ -234,6 +234,27 @@ class ThreadedEnginePerDevice : public ThreadedEngine {
     }
   }
 
+  /*!
+   * \brief Get number of cores this engine should reserve for its own use
+   * \param using_gpu Whether there is GPU usage
+   * \return number of cores that this engine wishes to be reserved
+   * \note Testing found no degradation of performance using these values
+   *       running cifar10 with resnet50 on various GPU systems,
+   *       including AWS p2.16xlarge, which has 16 GPU's
+   */
+  int GetReserveCoreCount(const bool using_gpu) const {
+    int reserve = 0;
+    if (using_gpu) {
+      // Save at least one for GPU tasks
+      ++reserve;
+      // If we have 8 or more real cores, reserve another core for GPU tasks
```

Comment:
Are 2 cores enough for 8 to 16 GPUs?

Reply:
I tried cifar10 with res50 on a p2.16xlarge and it had same images/sec as master branch. If you want a couple of extra reserved -- at least in the short term -- that's fine.

Reply:
Do you know of a workflow or model that will be using OMP in parallel or very close to GPU launches?  Generally I have see low CPU usage when training a GPU model (resnet50 in this case, also mnist).

Reply:
ok